From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Thu, 6 Aug 2026 12:37:37 +0200
Subject: refactor(net-lib): use strip_non_digits() to validate iSCSI LUN
 parameters

This function is a generalization of `_validate_iscsi_lun()`. It checks if the
first argument contains only digits; otherwise, it removes any non-digit
characters and prints a warning message.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/b4f5fe23003dfc7f11449ccdd3b67cc115fbb029
---
 modules.d/45net-lib/net-lib.sh | 25 ++++++++++---------------
 1 file changed, 10 insertions(+), 15 deletions(-)

diff --git a/modules.d/45net-lib/net-lib.sh b/modules.d/45net-lib/net-lib.sh
index c3842bf..1c9ff7d 100755
--- a/modules.d/45net-lib/net-lib.sh
+++ b/modules.d/45net-lib/net-lib.sh
@@ -365,19 +365,14 @@ normalized_iscsi_name() {
     printf "%s" "$1" | tr '[:upper:]' '[:lower:]' | tr -cd 'a-z0-9.:-'
 }
 
-_validate_iscsi_lun() {
-    local raw_lun="$1"
-
-    case "$raw_lun" in
-        *[!0-9]*)
-            warn "Given iSCSI LUN '$raw_lun' contains non-digits. Stripping non-digit characters."
-            printf "%s" "$raw_lun" | tr -cd '[:digit:]'
-            ;;
-        *)
-            # Purely decimal digits
-            echo "$raw_lun"
-            ;;
-    esac
+strip_non_digits() {
+    local str="$1"
+    local msg="$2"
+    if [ -n "$str" ] && ! isdigit "$str"; then
+        warn "${msg:+$msg }'$str' contains non-digits. Stripping non-digit characters."
+        str="$(printf "%s" "$str" | tr -cd '[:digit:]')"
+    fi
+    printf "%s" "$str"
 }
 
 parse_iscsi_root() {
@@ -464,7 +459,7 @@ parse_iscsi_root() {
             iscsi_netdev_name=$1
             shift
         fi
-        iscsi_lun=$(_validate_iscsi_lun "$1")
+        iscsi_lun=$(strip_non_digits "$1" "iSCSI LUN")
         shift
         if [ $# -ne 0 ]; then
             warn "Invalid parameter in iscsi: parameter!"
@@ -482,7 +477,7 @@ parse_iscsi_root() {
         fi
     fi
 
-    iscsi_lun=$(_validate_iscsi_lun "$1")
+    iscsi_lun=$(strip_non_digits "$1" "iSCSI LUN")
     shift
 
     iscsi_target_name=$(printf "%s:" "$@")
